Sources/ClientRuntime/Networking/Http/URLSession/URLSessionHTTPClient.swift

```diff
@@ -425,42 +457,108 @@ public final class URLSessionHTTPClient: HTTPClient {
-                dataTask.resume()
-                logger.info("start URLRequest(\(urlRequest.url?.absoluteString ?? "")) called")
-                Task { await streamBridge?.open() }
-            } catch {
-                continuation.resume(throwing: error)
+                // If needed, create a stream bridge that streams data from a SDK stream to a Foundation InputStream
+                // that URLSession can stream its request body from.
+                // Allow 16kb of in-memory buffer for request body streaming
+                // START - smithy.client.http.connections.acquire_duration
+                let acquireConnectionStart = Date().timeIntervalSinceReferenceDate
+                let streamBridge = requestStream.map {
+                    FoundationStreamBridge(
+                        readableStream: $0,
+                        bridgeBufferSize: 16_384,
+                        logger: logger,
+                        telemetry: telemetry,
+                        serverAddress: URLSessionHTTPClient.makeServerAddress(request: request))
+                }
+                let acquireConnectionEnd = Date().timeIntervalSinceReferenceDate
+                telemetry.connectionsAcquireDuration.record(
+                    value: acquireConnectionEnd - acquireConnectionStart,
+                    attributes: Attributes(),
+                    context: telemetryContext)
+                // END - smithy.client.http.connections.acquire_duration
+                let queuedEnd = acquireConnectionEnd
+                telemetry.requestsQueuedDuration.record(
+                    value: queuedEnd - queuedStart,
+                    attributes: Attributes(),
+                    context: telemetryContext)
+                // END - smithy.client.http.requests.queued_duration
+
+                // TICK - smithy.client.http.connections.limit
+                telemetry.httpMetricsUsage.connectionsLimit = session.configuration.httpMaximumConnectionsPerHost
+
+                // TICK - smithy.client.http.connections.usage
+                // TODO: instead of the transient stores, should rely on the Key/Value observer patttern
+                let totalCount = session.delegateQueue.operationCount
+                let maxConcurrentOperationCount = session.delegateQueue.maxConcurrentOperationCount
+                telemetry.httpMetricsUsage.acquiredConnections = totalCount < maxConcurrentOperationCount
+                    ? totalCount
+                    : maxConcurrentOperationCount
+                telemetry.httpMetricsUsage.idleConnections = totalCount - telemetry.httpMetricsUsage.acquiredConnections
+
+                // TICK - smithy.client.http.requests.usage
+                telemetry.httpMetricsUsage.inflightRequests = telemetry.httpMetricsUsage.acquiredConnections
+                telemetry.httpMetricsUsage.queuedRequests = telemetry.httpMetricsUsage.idleConnections
```

Comment:
These metrics are tricky, and should be investigated if there is a better way to observe the delegate queue's connections